substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -1223,12 +1242,25 @@ pub mod pallet {
 		#[pallet::weight(T::WeightInfo::set_payee())]
 		pub fn set_payee(
 			origin: OriginFor<T>,
-			payee: RewardDestination<T::AccountId>,
+			payee: PayoutDestination<T::AccountId>,
```

Comment:
What you could do here is to create a "newtype" like `struct CheckedPayoutDestination(PayoutDestination)`. This will have the same encoding and derives and everything, but you make the storage item only accept `CheckedPayoutDestination`.

Textbook example of how you use the Rust compile to catch errors for you at compile time, not runtime (ie. remembering that you have to do `PayoutDestination::from_call`).